core/arch/arm/plat-stm32mp1/scmi_server.c

```diff
@@ -374,6 +418,39 @@ const uint8_t *plat_scmi_protocol_list(unsigned int channel_id __unused)
 	return plat_protocol_list;
 }
 
+static bool nsec_can_access_resource(unsigned int fw_id)
+{
+	static struct firewall_controller *etzpc_fw_ctrl;
+	uint32_t query_arg = DECPROT(fw_id, DECPROT_NS_RW, DECPROT_UNLOCK);
+	struct firewall_query query = {
+		.arg_count = 1,
+		.args = &query_arg,
+		.ctrl = etzpc_fw_ctrl,
+	};
+
+	if (fw_id == STM32MP1_ETZPC_MAX_ID)
+		return true;
+
+	if (!etzpc_fw_ctrl) {
+		struct dt_driver_provider *prov = NULL;
+		int node = 0;
+
+		node = fdt_node_offset_by_compatible(get_embedded_dt(), -1,
```

Comment:
This is fine because we call the function at boot so we do not use this code at runtime but it's because we synchronize clocks in `stm32mp1_init_scmi_server()`

Reply:
Indeed, we get the firewall controller reference (handle) at boot time to be able to use it a run time. You you prefer I explicitly state that in an inline function here?

Reply:
I let you decide what's best. No strong opinion on what's clearer